### PR TITLE
Don't check commit messages of cherry-picked commits

### PR DESCRIPTION
**Pull request by zjhuntin**

### CODE REVIEW

**Comment by zjhuntin** [comment]
@akofink it will stop it from being added to the redmine issue now.

File: github/pull_request.rb [hunk 1]
```diff
@@ -82,6 +86,11 @@ def check_commits_style
       return
     end
 
+    if cherry_pick?
+       add_status('success', "PR is a Cherry-pick; commit message style not checked")
+       return
```
Comment (akofink) [inline github/pull_request.rb]: unnecessary? right?

File: github/pull_request.rb [hunk 2]
```diff
@@ -61,7 +65,7 @@ def commits
   end
 
   def issue_numbers
-    if @issue_numbers.nil?
+    if @issue_numbers.nil?
```
Comment (akofink) [inline github/pull_request.rb]: random trailing whitespace. No rubocop here to tell us that.

File: github/pull_request.rb (hunk 1 above)
Comment (akofink) [reply github/pull_request.rb]: Oh, nvm. Ignore me. :)

File: github/pull_request.rb [hunk 3]
```diff
@@ -128,7 +137,7 @@ def check_commits_style
   end
 
   def add_issue_links
-    if new? && issue_numbers.any?
+    if new? && issue_numbers.any? && !cherry_pick?
```
Comment (ekohl) [inline github/pull_request.rb]: I think this is not needed. If the PR does include both issue numbers and CPs then it's fine to link those that are included. If there aren't any, it's a noop anyway. It doesn't hurt either though.

File: test/github/pull_request_test.rb [hunk 4]
```diff
@@ -17,6 +17,20 @@ def test_wip_is_wip
       assert pr.wip?, title
     end
   end
+  
+  def test_wip_regular_title
+    ['The commit', 'The original commit'].each do |title|
```
Comment (ekohl) [inline test/github/pull_request_test.rb]: You could also test `the CP commit`

File: github/pull_request.rb (hunk 3 above)
Comment (akofink) [reply github/pull_request.rb]: So, are you saying it's fine to have cherry-pick PRs linked to the redmine issues (well, just the first commit's issue)? I find the behavior annoying and inconsistent. Maybe if it linked *all* the issues in a cherry-pick PR, it'd be fine, but it's just the first one. I'd rather just not have the extra links.

Comment (ekohl) [reply github/pull_request.rb]: Also fair. Since I usually don't use those links, I'm fine with whatever you decide.

Comment (akofink) [reply github/pull_request.rb]: @ekohl oh, right. :) I do want to disable adding links to redmine also @zjhuntin.

**Comment by zjhuntin** [comment]
@ekohl change made!

**Comment by ekohl** [comment]
Waiting for Travis to be green before merging.

**Comment by akofink** [comment]
Oh! :) :1st_place_medal:

**Comment by akofink** [comment]
I don't understand though... isn't it this line? https://github.com/theforeman/prprocessor/blob/1ea8acb1ca8dc9087111d1d1d0baf2d256bf30a2/app.rb#L72

**Comment by zjhuntin** [comment]
@akofink good catch, I thought that I had caught it before, but I updated. @ekohl do you think this change looks okay? Or should this be done in a different place on a different pull request?

File: app.rb [hunk 5]
```diff
@@ -69,7 +69,7 @@
           issue.set_release(nil)
         end
 
-        issue.add_pull_request(pull_request.raw_data['html_url'])
+	issue.add_pull_request(pull_request.raw_data['html_url']) if !pull_request.cherry_pick?
```
Comment (akofink) [inline app.rb]: Omg, why is that first indentation 3 spaces!? lol

This is fine, but more Rubyist to use `unless`.

File: github/pull_request.rb [hunk 6]
```diff
@@ -82,6 +86,11 @@ def check_commits_style
       return
     end
 
+    if cherry_pick?
+       add_status('success', "PR is a Cherry-pick; commit message style not checked")
```
Comment (ekohl) [inline github/pull_request.rb]: Could you change the indenting to 2 spaces rather than 3?

**Pull request merged**